### DISCUSSION TITLE
International customers

### DISCUSSION THREAD

**Discussion by CosmicMac** [FoloToy/folotoy-server-self-hosting#144] [General | 讨论]
Hello dear Chinese friends,

Folotoy products are amazing!
As a tinkerer, I'm really interested in the Octopus but unfortunately Taobao is not really targeted at international customers (I can't even pass the first captcha, even with automatic translation activated in my browser).

Do you have plan to sell your products on AliExpress?
Or maybe it's possible to direct order to your office?

Cheers from France!

**Comment by SHLcy** [comment]
> Hello dear Chinese friends,
> 
> Folotoy products are amazing! As a tinkerer, I'm really interested in the Octopus but unfortunately Taobao is not really targeted at international customers (I can't even pass the first captcha, even with automatic translation activated in my browser).
> 
> Do you have plan to sell your products on AliExpress? Or maybe it's possible to direct order to your office?
> 
> Cheers from France!

Could you provide us with your Email?

**Comment by lewangdev** [comment]
> Hello dear Chinese friends,
> 
> 
> 
> Folotoy products are amazing!
> 
> As a tinkerer, I'm really interested in the Octopus but unfortunately Taobao is not really targeted at international customers (I can't even pass the first captcha, even with automatic translation activated in my browser).
> 
> 
> 
> Do you have plan to sell your products on AliExpress?
> 
> Or maybe it's possible to direct order to your office?
> 
> 
> 
> Cheers from France!

I am Larry, the founder of FoloToy. Here is my email: lewang.dev@gmail.com

Please feel free to contact me.

**Reply by CosmicMac** [reply]
Done! Thanks 🙂

**Comment by CosmicMac** [comment]
Finally! I received my Octopus and I'm super happy with it...
I did quick tests and it runs 100% offline with [LocalAI](https://github.com/mudler/LocalAI) and [Ollama](https://github.com/ollama/ollama).
Next steps: get rid of Ollama (LocalAI does LLM too) and replace edge TTS with LocalAI/Bark.
BTW, I find it really handy to be able to make it say arbitrary things just by publishing MQTT messages. 👍

**Reply by lewangdev** [reply]
Thank you very much for sharing your user experience with us. We also noticed that LocalAI is an open-source alternative to the OpenAI interface, but we found that the URL of LocalAI's TTS interface is different from OpenAI's. It may be necessary to use Nginx for URL rewriting in order to make it work.

[LocalAI TTS Doc](https://localai.io/features/text-to-audio/)
```
curl http://localhost:8080/tts -H "Content-Type: application/json" -d '{
  "input": "Hello world",
  "model": "tts"
}'
```

**Reply by CosmicMac** [reply]
Actually the doc is a bit confusing as it says
> The /tts endpoint can **also** be used to generate speech from text  

But there is the classic `/v1` entry point exposed as well.
```
curl --location 'http://localhost:8080/v1/audio/speech' \
--header 'Content-Type: application/json' \
--data '{
    "model": "fr-siwis-low.onnx",
    "input": "Les sanglots longs des violons de l'\''automne blessent mon coeur d'\''une langueur monotone.",
    "response_format": "mp3",
    "speed": "1"
}'
```
I only scratched the surface but for the moment it seems to generate only wav files were FoloToy server expects mp3 (all files are saved with the mp3 extension, even if they are wav files).
I'll give it another try tonight.

**Reply by lewangdev** [reply]
Yes, we hardcoded the MP3 format without considering the "response_format" parameter. This issue will be resolved in the next version. Thank you.

**Reply by CosmicMac** [reply]
That makes sense as MP3 is supposed to be the default response format, according to [OpenAI API doc](https://platform.openai.com/docs/api-reference/audio/createSpeech).
So I guess LocalAI is not 100% compatible after all (at least when using Piper as the TTS backend, which seems to be limited to 16-bit mono PCM).
